app/uk/gov/hmrc/gform/formtemplate/FormTemplatesControllerRequestHandler.scala

```diff
@@ -164,6 +164,48 @@ object FormTemplatesControllerRequestHandler {
         Json.obj()
       )
 
+    val transformProgressIndicator = {
+      import uk.gov.hmrc.gform.core.parsers.LabelSizeParser
+      import uk.gov.hmrc.gform.sharedmodel.formtemplate.LabelSize
+
+      val pickProgressIndicatorSize = (__ \ 'progressIndicatorSize).json
+        .pick[JsString]
+        .map(x => LabelSizeParser.validate(x.value).right.toOption.getOrElse(Medium)) orElse Reads.pure(Medium)
+
+      val pickProgressIndicator = (__ \ 'progressIndicator).json.pick
+
+      val createProgressIndicator =
+        (pickProgressIndicator ~ pickProgressIndicatorSize).tupled.map(x =>
+          Json.obj("progressIndicator" -> Json.obj("label" -> x._1, "labelSize" -> LabelSize.format.writes(x._2)))
+        )
+
+      __.json.update(createProgressIndicator) andThen (__ \ 'progressIndicatorSize).json.prune orElse (__.read)
+
+    }
+
+    val transformProgressIndicatiorInArray =
+      __.read[JsArray].map { pages =>
+        JsArray(
+          pages.value.map(_.transform(transformProgressIndicator).get)
+        )
+      }
```

Comment:
This can be replaced by:

```suggestion
    val transformProgressIndicatiorInArray: Reads[JsArray] = Reads.list(transformProgressIndicator).map(JsArray.apply)
```